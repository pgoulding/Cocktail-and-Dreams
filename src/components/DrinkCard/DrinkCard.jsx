import React from 'react'
import './DrinkCard.scss'
import { connect } from 'react-redux'
import { addFavorite, removeFavorite } from '../../actions/index'

const DrinkCard = ({drinkInfo, ingredients, favoriteCocktails, addCocktail, removeCocktail}) => {
  const drinkCard = {...drinkInfo}

 const updateStore = (e) => {
    e.preventDefault()
    if (favoriteCocktails.find(favCocktails => favCocktails.drinkId === drinkCard.drinkId)) {
      removeCocktail(drinkInfo.drinkId)
      console.log(drinkCard.drinkId)
    } else {
     addCocktail(drinkCard)
    }
  }

  return (
    <div className="drink-card" >
      <h3>{drinkCard.drinkName}</h3>
      <div className="drink-card-info">
        <img className="drink-image" src={drinkCard.drinkImageSource} alt={`${drinkCard.drinkName}`} />
        <div>
          <p>{drinkCard.instructions}</p>
        </div>
      </div>
      <div className='drink-card-instructions'>
        <ul>
        {drinkCard.ingredients.map(item => {
          if (item.ingredient) {
            return <li 
            className={ingredients.find(name => name.strIngredient === item.ingredient) ? 'found-ingredient': 'no-ingredient'}
            >{item.ingredient} : {item.amount}</li>
          }
        })}
        </ul>
        <div>
          <p>Type: {drinkCard.alcoholic}</p>
          <p>Glass: {drinkCard.glass}</p>
        </div>
      </div>
      <div>
        <button onClick={(e) => updateStore(e)}>
          Favorite
          {/* {fav ? 'Remove Favorite': ' Add Favorite'} */}
        </button>
      </div>
    </div>
  ) 
}

const mapStateToProps = ({ingredients, favoriteCocktails}) => ({
  ingredients,
  favoriteCocktails
})

const mapDispatchToProps = (dispatch) => ({
  addCocktail: (cocktail) => dispatch(addFavorite(cocktail)),
  removeCocktail: (id) => dispatch(removeFavorite(id))
})


export default connect(mapStateToProps, mapDispatchToProps)(DrinkCard)
