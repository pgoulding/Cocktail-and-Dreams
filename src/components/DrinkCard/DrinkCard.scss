.drink-card {
  max-width: 80%;
  background-color: lightblue;
  padding: 15px;
  margin: 20px;
  border-radius: 4px;
  h3{
    text-align: center;
  }
  .drink-card-info{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    p{
      margin:0 10px;
    }
  }
  .drink-image{
    height: 200px;
  }
  .drink-card-instructions{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }
  ul{
    list-style: none;
    img{
      height:1.2rem;
    }
    .no-ingredient {
      text-decoration: line-through;
    }
  }
}