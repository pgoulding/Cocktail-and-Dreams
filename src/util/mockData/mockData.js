export const ingredientsList = [
  {"strIngredient1": "Dark rum"},
  {"strIngredient1": "Sweet Vermouth"},
  {"strIngredient1": "Strawberry schnapps"},
  {"strIngredient1": "Scotch"},
  {"strIngredient1": "Apricot brandy"},
  {"strIngredient1": "Triple sec"},
  {"strIngredient1": "Apple brandy"},
  {"strIngredient1": "Carbonated water"},
  {"strIngredient1": "Chocolate syrup"},
]

export const mockUserIngredients =[
  {
    "idIngredient": "571",
    "strIngredient": "151 proof rum",
    "strDescription": "Bacardi 151 is a brand of highly alcoholic rum made by Bacardi Limited of Hamilton, Bermuda.",
    "strType": "Rum"
  },
{
    "idIngredient": "573",
    "strIngredient": "Absinthe",
    "strDescription": "Absinthe is historically described as a distilled, highly alcoholic (45–74% ABV / 90–148 U.S. proof) beverage.[1][2][3][4] It is an anise-flavoured spirit derived from botanicals, including the flowers and leaves of Artemisia absinthium (\"grand wormwood\"), together with green anise, sweet fennel, and other medicinal and culinary herbs.",
    "strType": "Spirit"
  },
  {
    "idIngredient": "8",
    "strIngredient": "Absolut Vodka",
    "strDescription": "Absolut Vodka is a brand of vodka, produced near Åhus, in southern Sweden.",
    "strType": "Vodka"
  },
{
    "idIngredient": "18",
    "strIngredient": "Amaretto",
    "strDescription": "Amaretto (Italian for \"a little bitter\") is a sweet, almond-flavoured, Italian liqueur associated with Saronno, Italy. Various commercial brands are made from a base of apricot pits, almonds, or both. When served as a beverage.",
    "strType": "Liqueur"
  },
{
    "idIngredient": "221",
    "strIngredient": "Fruit Punch",
    "strDescription": null,
    "strType": null
  },
{
    "idIngredient": "457",
    "strIngredient": "Sour Mix",
    "strDescription": "Sour mix (also known as sweet and sour mix) is a mixer that is yellow-green in color and is used in many cocktails",
    "strType": "Soft Drink"
  }
]

export const mockDrink = {
  "idDrink": "11007",
  "strDrink": "Margarita",
  "strAlcoholic": "Alcoholic",
  "strGlass": "Cocktail glass",
  "strInstructions": "Rub the rim of the glass with the lime slice to make the salt stick to it. Take care to moisten only the outer rim and sprinkle the salt on it. The salt should present to the lips of the imbiber and never mix into the cocktail. Shake the other ingredients with ice, then carefully pour into the glass.",
  "strDrinkThumb": "https://www.thecocktaildb.com/images/media/drink/wpxpvu1439905379.jpg",
  "strIngredient1": "Tequila",
  "strIngredient2": "Triple sec",
  "strIngredient3": "Lime juice",
  "strMeasure1": "1 1/2 oz ",
  "strMeasure2": "1/2 oz ",
  "strMeasure3": "1 oz ",
  "strCreativeCommonsConfirmed": "No",
  "dateModified": "2015-08-18 14:42:59"
}

export const mockCleanedDrink = {
  drinkId: "11007",
  drinkName: "Margarita",
  alcoholic: "Alcoholic",
  glass: "Cocktail glass",
  instructions: "Rub the rim of the glass with the lime slice to make the salt stick to it. Take care to moisten only the outer rim and sprinkle the salt on it. The salt should present to the lips of the imbiber and never mix into the cocktail. Shake the other ingredients with ice, then carefully pour into the glass.",
  favorited: false,
  drinkImageSource: "https://www.thecocktaildb.com/images/media/drink/wpxpvu1439905379.jpg",
  ingredients:[
    {ingredient: "Tequila", amount:"4oz"},
    {ingredient:"Mix", amount:"8oz"},
  ]
}