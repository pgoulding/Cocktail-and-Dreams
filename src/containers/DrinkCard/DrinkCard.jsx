import React from 'react'
import './DrinkCard.scss'
import { connect } from 'react-redux'
import { addFavorite, removeFavorite } from '../../actions/index'

export const DrinkCard = ({drinkInfo, ingredients, favoriteCocktails, addCocktail, removeCocktail}) => {
  const drinkCard = {...drinkInfo}

 const updateStore = (e) => {
    e.preventDefault()
    if (favoriteCocktails.find(favCocktails => favCocktails.drinkId === drinkCard.drinkId)) {
      removeCocktail(drinkInfo.drinkId)
    } else {
      addCocktail(drinkCard)
    }
  }

  return (
    <div className="drink-card" >
      <h3>{drinkCard.drinkName}</h3>
      <div className="drink-card-info">
        <img className="drink-image" src={drinkCard.drinkImageSource} alt={`${drinkCard.drinkName}`} />
        <div>
          <p>{drinkCard.instructions}</p>
        </div>
      </div>
      <div className='drink-card-instructions'>
        <ul>
        {drinkCard.ingredients.map(item => {
          if (item.ingredient) {
            return <li 
            className={ingredients.find(name => name.strIngredient.toLowerCase() === item.ingredient.toLowerCase()) ? 'found-ingredient': 'no-ingredient'}
            key={item.amount+Date.now()}>{item.ingredient} : {item.amount}</li>
          }
        })}
        </ul>
        <div>
          <p>Type: {drinkCard.alcoholic}</p>
          <p>Glass: {drinkCard.glass}</p>
        </div>
      </div>
      <div>
        <button className='favorite-button' onClick={(e) => updateStore(e)}>
          Favorite
          {/* {fav ? 'Remove Favorite': ' Add Favorite'} */}
        </button>
      </div>
    </div>
  ) 
}

export const mapStateToProps = ({ingredients, favoriteCocktails}) => ({
  ingredients,
  favoriteCocktails
})

export const mapDispatchToProps = (dispatch) => ({
  addCocktail: (cocktail) => dispatch(addFavorite(cocktail)),
  removeCocktail: (id) => dispatch(removeFavorite(id))
})


export default connect(mapStateToProps, mapDispatchToProps)(DrinkCard)
