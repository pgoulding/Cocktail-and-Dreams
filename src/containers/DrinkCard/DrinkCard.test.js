import React from 'react';
import { shallow } from 'enzyme';
import { DrinkCard } from './DrinkCard';
import { mockDrink, mockIngredients, mockUserIngredients, mockCleanedDrink } from '../../util/mockData/mockData'
describe('Drink Card', () => {

  it('should match snapshot', () => {
    let wrapper = shallow(<DrinkCard drinkInfo={mockCleanedDrink} ingredients={mockUserIngredients} favoriteCocktails={[mockDrink]} addCocktail={jest.fn()} removeCocktail={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  });

});