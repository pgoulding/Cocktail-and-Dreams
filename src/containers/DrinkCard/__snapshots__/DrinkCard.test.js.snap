// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drink Card should match snapshot 1`] = `ShallowWrapper {}`;
